fe/fe-core/src/main/java/com/starrocks/analysis/ColumnIdExpr.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    public Expr convertToColumnNameExpr(Map<ColumnId, Column> idToColumn) {
+        Expr clonedExpr = expr.clone();
+        setColumnName(idToColumn, clonedExpr);
+        return clonedExpr;
+    }
+
+    public Expr convertToColumnNameExpr(List<Column> schema) {
+        Expr clonedExpr = expr.clone();
+        setColumnName(MetaUtils.convertToIdToColumn(schema), clonedExpr);
+        return clonedExpr;
+    }
+
+    public Expr getExpr() {
+        return expr;
+    }
+
+    public String serialize() {
+        return new ExprSerializeVisitor().visit(expr);
+    }
+
+    public static ColumnIdExpr deserialize(String sql) {
+        Expr expr = SqlParser.parseSqlToExpr(sql, SqlModeHelper.MODE_DEFAULT);
+        setColumnIdToColumnName(expr);
+        return new ColumnIdExpr(expr);
+    }
+
+    private void setColumnName(Map<ColumnId, Column> idToColumn, Expr expr) {
+        if (expr instanceof SlotRef) {
+            SlotRef slotRef = (SlotRef) expr;
+            Column column = idToColumn.get(slotRef.getColumnId());
+            if (column == null) {
+                throw new SemanticException(String.format("can not get column by column id: %s", slotRef.getColumnId()));
+            }
+            slotRef.setColumnName(column.getName());
+        }
+
+        for (Expr child : expr.getChildren()) {
+            setColumnName(idToColumn, child);
+        }
+    }
+
+    private static void setColumnId(Map<String, Column> nameToColumn, Expr expr) {
+        if (expr instanceof SlotRef) {
+            SlotRef slotRef = (SlotRef) expr;
+            Column column = nameToColumn.get(slotRef.getColumnName());
+            if (column == null) {
+                throw new SemanticException(String.format("can not get column by name : %s", slotRef.getColumnName()));
+            }
+            slotRef.setColumnId(column.getColumnId());
+        }
+
+        for (Expr child : expr.getChildren()) {
+            setColumnId(nameToColumn, child);
+        }
+    }
+
+    private static void setColumnIdToColumnName(Expr expr) {
+        if (expr instanceof SlotRef) {
+            SlotRef slotRef = (SlotRef) expr;
+            slotRef.setColumnId(ColumnId.create(slotRef.getColumnName()));
+        }
+
+        for (Expr child : expr.getChildren()) {
+            setColumnIdToColumnName(child);
+        }
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj.getClass() != ColumnIdExpr.class) {
+            return false;
+        }
+
+        return this.expr.equals(((ColumnIdExpr) obj).expr);
+    }
+
+    private static class ExprSerializeVisitor extends AstToStringBuilder.AST2StringBuilderVisitor {
+        @Override
+        public String visitSlot(SlotRef node, Void context) {
+            if (node.getTblNameWithoutAnalyzed() != null) {
+                return node.getTblNameWithoutAnalyzed().toString() + "." + node.getColumnId().getId();
```

Comment:
why use column id here, should use column name?

Reply:
this is used when ColumnIdExpr is serializing to store, we should use ColumnId.